packages/ethereum-light-client/src/config.rs

```diff
@@ -0,0 +1,382 @@
+use core::{
+    fmt::{self, Debug},
+    str::FromStr,
+};
+
+use serde::{Deserialize, Serialize};
+use typenum::{NonZero, Unsigned};
+
+use alloy_primitives::FixedBytes;
+
+/// Minimal config.
+#[derive(Debug, Clone, PartialEq, Eq, Default)]
+pub struct Minimal;
+
+/// Mainnet config.
+#[derive(Debug, Clone, PartialEq, Eq, Default)]
+pub struct Mainnet;
+
+#[derive(Debug, Clone, Copy, PartialEq, Eq, PartialOrd, Ord, Serialize, Deserialize)]
+#[serde(rename_all = "snake_case")]
+pub enum PresetBaseKind {
+    Minimal,
+    Mainnet,
+}
+
+impl fmt::Display for PresetBaseKind {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        f.write_str(match self {
+            PresetBaseKind::Minimal => "minimal",
+            PresetBaseKind::Mainnet => "mainnet",
+        })
+    }
+}
+
+impl FromStr for PresetBaseKind {
+    type Err = String;
```

Comment:
Better error type here?

Reply:
I'll look into it. This file was really just a copy paste from Union because it made my life a lot easier to have around. Pretty sure I will change that whole thing at some point.

Reply:
I used anyhow::Result everywhere lol. So not a big deal for now

Reply:
Yeah, we can look into getting a cohesive error story for the all the rust code at some point. I guess it's not critical right now :)